test/e2e/operator.go

```diff
@@ -419,3 +420,131 @@ var _ = Describe("ARO Operator - MHC Deployment", func() {
+		blockList := imageconfig.Spec.RegistrySources.BlockedRegistries
+
+		return sliceEqual(allowList, expectedAllowlist) && sliceEqual(blockList, expectedBlocklist), nil
+	}
+
+	AfterEach(func() {
+		// Clear allow/block lists
+		imageconfig, err := clients.ConfigClient.ConfigV1().Images().Get(ctx, "cluster", metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		imageconfig.Spec.RegistrySources.AllowedRegistries = nil
+		imageconfig.Spec.RegistrySources.BlockedRegistries = nil
+		expectedAllowlist = nil
+		expectedBlocklist = nil
+
+		_, err = clients.ConfigClient.ConfigV1().Images().Update(ctx, imageconfig, metav1.UpdateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		Eventually(verifyLists).WithTimeout(timeout).Should(BeTrue())
+	})
+
+	It("should have empty allow and block lists by default", func() {
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, "cluster", metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		if !instance.Spec.OperatorFlags.GetSimpleBoolean(imageconfigFlag) {
+			Skip("ImageConfig Controller is not enabled, skipping test")
+		}
+
+		imageconfig, err := clients.ConfigClient.ConfigV1().Images().Get(ctx, "cluster", metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		allowList := imageconfig.Spec.RegistrySources.AllowedRegistries
+		blockList := imageconfig.Spec.RegistrySources.BlockedRegistries
+
+		Expect(allowList).To(BeEmpty())
+		Expect(blockList).To(BeEmpty())
+	})
+
+	It("should add registries to the allow list", func() {
+
+		// Check that Image Config is enabled
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, "cluster", metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		if !instance.Spec.OperatorFlags.GetSimpleBoolean(imageconfigFlag) {
+			Skip("ImageConfig Controller is not enabled, skipping test")
+		}
+
+		// Get required registries for this cluster
+		requiredRegistries := getLocationRegistries(instance.Spec.ACRDomain, "azurecr.io", instance.Spec.Location)
+		expectedAllowlist = append(requiredRegistries, optionalRegistry)
+
+		// Get current imageconfig object
+		imageconfig, err := clients.ConfigClient.ConfigV1().Images().Get(ctx, "cluster", metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Update allow list
+		imageconfig.Spec.RegistrySources.AllowedRegistries = append(imageconfig.Spec.RegistrySources.AllowedRegistries, optionalRegistry)
+
+		_, err = clients.ConfigClient.ConfigV1().Images().Update(ctx, imageconfig, metav1.UpdateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		Eventually(verifyLists).WithTimeout(timeout).Should(BeTrue())
+	})
+
+	It("should add optional registries to the block list and skip required registries", func() {
+
+		// Check that Image Config is enabled
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, "cluster", metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		if !instance.Spec.OperatorFlags.GetSimpleBoolean(imageconfigFlag) {
```

Comment:
Could we do this check once at the beginning of the test spec rather than twice?

Reply:
After looking over the [Ginkgo documentation](https://onsi.github.io/ginkgo/#skipping-specs) it does not appear there is any way to skip a container node (`Describe()` and `When()` blocks), only individual test nodes (`It()` and `Specify()`), and the skipping needs to be done inside the test block itself, not in a `Before()`.

Reply:
After looking over the [Ginkgo documentation](https://onsi.github.io/ginkgo/#skipping-specs) it does not appear there is any way to skip a container node (`Describe()` and `When()` blocks), only individual test nodes (`It()` and `Specify()`), and the skipping needs to be done inside the test block itself, not in a `Before()`. It _could_ be moved to a `BeforeEach()` but that would still result in the flag being checked three times. It would simplify the code though, so it's a good idea.

Reply:
Configuration extracted.